src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -633,7 +641,27 @@ public CSharpCompilation WithScriptCompilationInfo(CSharpScriptCompilationInfo?
                 isSubmission: info != null,
                 _referenceManager,
                 reuseReferenceManager,
-                syntaxAndDeclarations: _syntaxAndDeclarations);
+                _syntaxAndDeclarations,
+                this.SemanticModelProvider);
+        }
+
+        /// <summary>
+        /// Returns a new compilation with the given semantic model provider.
+        /// </summary>
+        internal override Compilation WithSemanticModelProvider(SemanticModelProvider? semanticModelProvider)
+        {
+            return new CSharpCompilation(
```

Comment:
>return new CSharpCompilation( [](start = 12, length = 29)

Missing a check to see if we already using this provider?

Reply:
Yes, sure.